sycl/source/detail/graph_impl.hpp

```diff
@@ -106,7 +106,11 @@ class node_impl {
 
   /// Query if this is an empty node.
   /// @return True if this is an empty node, false otherwise.
-  bool isEmpty() const { return MIsEmpty; }
+  bool isEmpty() const {
+    if (MCGType == sycl::detail::CG::None)
+      return true;
+    return MIsEmpty;
+  }
```

Comment:
I know I suggested this originally, but I think it would actually be preferable to remove `MIsEmpty` from the class altogether and just set `MCGType = sycl::detail::CG::None` in the constructor for empty nodes, then we can simplify this to just:

```c++
  bool isEmpty() const {
    return MCGType == sycl::detail::CG::None;
  }
```